File: swift/WisImplementationExample/Models/VehicleType.swift
//
//  VehicleType.swift
//  WisImplementationExample
//
//  Created by Bertrand VILLAIN on 18/10/2021.
//

import Foundation

enum VehicleType: String, Codable {
  
  case car
  case truck
  case pedestrian
  case bus
  case motorcycle
  case bicycle

  init(from decoder: Decoder) throws {
    let container = try decoder.singleValueContainer()
    let rawValue = try container.decode(String.self)
    guard let state = VehicleType(rawValue: rawValue) else { throw WisNetworkError.errorParsingJSON("VehicleType") }
    self = state
  }
  
}

File: android/app/src/main/java/io/wassa/wisexampleproject/model/VehicleType.kt
package io.wassa.wisexampleproject.model

import com.google.gson.annotations.SerializedName

enum class VehicleType {
    @SerializedName("car")
    Car,
    @SerializedName("truck")
    Truck,
    @SerializedName("pedestrian")
    Pedestrian,
    @SerializedName("bus")
    Bus,
    @SerializedName("motorcycle")
    Motorcycle,
    @SerializedName("bicycle")
    Bicycle
}